module/x/gravity/keeper/batch.go

```diff
@@ -81,7 +82,22 @@ func (k Keeper) batchTxExecuted(ctx sdk.Context, tokenContract common.Address, n
 		}
 		return false
 	})
+
+	// burn the amount for non cosmos originated asset
+	isCosmosOriginated, denom := k.ERC20ToDenomLookup(ctx, common.HexToAddress(batchTx.TokenContract))
+	if !isCosmosOriginated {
+		totalToBurn := sdk.NewInt(0)
+		for _, tx := range batchTx.Transactions {
```

Comment:
just some sanity check -- all txs in a batch are of the same contract/denom?
and ok to burn if the tx is invalid / fails later?

Reply:
I added an extra sanity check.

It is okay to burn even if the tx fails on the contract level (reverted) because they are still considered as "transfered" on chain as we create a voucher for them to redeem the failed tokens in the contract level